hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/SnapshotModeOp.java

```diff
@@ -0,0 +1,510 @@
+     * time at which the test is run.
+     *
+     * <p>Two {@link GeneratedMessageV3} messages are fuzzy-equal iff they have the same number of fields, where each
+     * un-skipped primitive field matches exactly; and each un-skipped list field consists of fuzzy-equal elements.
+     *
+     * @param expectedMessage the expected message
+     * @param expectedPlaceholderNum the placeholder number for the expected message
+     * @param actualMessage the actual message
+     * @param actualPlaceholderNum the placeholder number for the actual message
+     * @param mismatchContext a supplier of a string that describes the context of the mismatch
+     */
+    private static void fuzzyMatch(
+            @NonNull GeneratedMessageV3 expectedMessage,
+            final long expectedPlaceholderNum,
+            @NonNull GeneratedMessageV3 actualMessage,
+            final long actualPlaceholderNum,
+            @NonNull final Supplier<String> mismatchContext) {
+        final var expectedType = expectedMessage.getClass();
+        final var actualType = actualMessage.getClass();
+        if (!expectedType.equals(actualType)) {
+            Assertions.fail("Mismatched types between expected " + expectedType + " and " + actualType + " - "
+                    + mismatchContext.get());
+        }
+        expectedMessage = normalized(expectedMessage, expectedPlaceholderNum);
+        actualMessage = normalized(actualMessage, actualPlaceholderNum);
+        // getAllFields() returns a TreeMap so ordering is deterministic here
+        final var expectedFields =
+                new ArrayList<>(expectedMessage.getAllFields().entrySet());
+        final var actualFields = new ArrayList<>(actualMessage.getAllFields().entrySet());
+        if (expectedFields.size() != actualFields.size()) {
+            Assertions.fail("Mismatched field counts between expected " + expectedMessage + " and " + actualMessage
+                    + " - " + mismatchContext.get());
+        }
+        for (int i = 0, n = expectedFields.size(); i < n; i++) {
+            final var expectedField = expectedFields.get(i);
+            final var actualField = actualFields.get(i);
+            final var expectedName = expectedField.getKey().getName();
+            final var actualName = actualField.getKey().getName();
+            if (!Objects.equals(expectedName, actualName)) {
+                Assertions.fail(
+                        "Mismatched field names ('" + expectedName + "' vs '" + actualName + "' between expected "
+                                + expectedMessage + " and " + actualMessage + " - " + mismatchContext.get());
+            }
+            if (FIELDS_TO_SKIP_IN_FUZZY_MATCH.contains(expectedName)) {
+                continue;
+            }
+            final var expectedValue = expectedField.getValue();
+            final var actualValue = actualField.getValue();
+            if (expectedValue instanceof List<?> expectedList) {
```

Comment:
these nested `if`s look so much like `matchValue` I'm wondering if L355-L384 should be broken out to a separate method to make that more obvious?  

probably doesn't matter

Reply:
Renamed `matchValue()` to `matchSingleValue()` and extracted this as `matchValue()`.